lib/c-api/src/wasm_c_api/mod.rs

```diff
@@ -208,6 +208,26 @@ pub unsafe extern "C" fn wasm_module_new(
     }))
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn wasm_module_validate(
+    store_ptr: Option<NonNull<wasm_store_t>>,
+    bytes: &wasm_byte_vec_t,
+) -> bool {
+    // TODO: review lifetime of byte slice
+    let wasm_byte_slice: &[u8] = slice::from_raw_parts_mut(bytes.data, bytes.size);
+
+    if store_ptr.is_none() {
+        return false;
+    }
+
+    let store_ptr = store_ptr.unwrap().cast::<Store>();
+    let store = store_ptr.as_ref();
+
+    c_try!(Module::validate(store, wasm_byte_slice); otherwise return false);
+
+    true
```

Comment:
I think this code should be able to just be

```Rust
     c_try!(Module::validate(store, wasm_byte_slice); otherwise return false)
```

Reply:
Nah, because `Module::validate` returns a `Result`.

Reply:
I thought that part of `c_try` was for things that returned `Result`